programs/validator-history/src/instructions/copy_cluster_info.rs

```diff
@@ -0,0 +1,56 @@
+use anchor_lang::{
+    prelude::*,
+    solana_program::{clock::Clock, slot_history::Check},
+};
+
+use crate::{errors::ValidatorHistoryError, utils::cast_epoch, ClusterHistory};
+
+#[derive(Accounts)]
+pub struct CopyClusterInfo<'info> {
+    #[account(
+        mut,
+        seeds = [ClusterHistory::SEED],
+        bump,
+    )]
+    pub cluster_history_account: AccountLoader<'info, ClusterHistory>,
+    pub slot_history: UncheckedAccount<'info>,
+    #[account(mut)]
+    pub signer: Signer<'info>,
+}
+
+pub fn handler(ctx: Context<CopyClusterInfo>) -> Result<()> {
```

Comment:
we should add support for current epoch too. if validator doing extremely poorly for current epoch, stakenet should be able to pick that up and undelegate

Reply:
added extra call for setting current epoch, and cluster_history_last_update_slot field to top level struct